### PR TITLE
Fixes to source generator documentation

### PR DESCRIPTION
**Pull request by agocke**
/cc @cston @dotnet/roslyn-compiler

### CODE REVIEW

File: docs/features/generators.md [hunk 1]
```diff
@@ -24,8 +25,14 @@ Source generators are implementations of `Microsoft.CodeAnalysis.SourceGenerator
...
 ```
 `SourceGenerator` implementations are defined in external assemblies passed to the compiler
-using the same `-analyzer:` option used for diagnostic analyzers. An assembly can
-contain a mix of diagnostic analyzers and source generators.
+using the same `-analyzer:` option used for diagnostic analyzers. Valid source generators
+must:
+
+1. Implement `Microsoft.CodeAnalysis.SourceGenerator`
+1. Be decorated with the `Microsoft.CodeAnalysis.SourceGeneratorAttribute` to indicate 
+   supported languages.
```
Comment (cston) [inline docs/features/generators.md]: Thanks. Perhaps also add a note about `-feature`, something like: Source generators are currently an experimental feature in the compiler. To enable, include `-feature:replace` on the command line or ```<Features>replace</Features>``` in the `.proj`.

**Comment by cston** [comment]
LGTM

File: docs/features/generators.md [hunk 2]
```diff
@@ -7,12 +7,16 @@ Source generators provide a mechanism through which source code can be generated
 and added to the compilation. The additional source can be based on the content of the compilation,
 enabling some meta-programming scenarios.
 
+Like all pre-release features, code generators will need to be specially enabled, either by passing
+`/features:replace` directly to `csc` or by setting the `<Features>replace</Features>` flag in the project file.
```
Comment (jcouv) [inline docs/features/generators.md]: Which part of the project file should `<Feature>` node go into?

File: docs/features/generators.md [hunk 3]
```diff
@@ -24,8 +28,14 @@ Source generators are implementations of `Microsoft.CodeAnalysis.SourceGenerator
     }
 ```
 `SourceGenerator` implementations are defined in external assemblies passed to the compiler
-using the same `-analyzer:` option used for diagnostic analyzers. An assembly can
-contain a mix of diagnostic analyzers and source generators.
+using the same `-analyzer:` option used for diagnostic analyzers. Valid source generators
+must:
+
+1. Implement `Microsoft.CodeAnalysis.SourceGenerator`
+1. Be decorated with the `Microsoft.CodeAnalysis.SourceGeneratorAttribute` to indicate
```
Comment (jcouv) [inline docs/features/generators.md]: Bullet should be 2

Comment (agocke) [inline docs/features/generators.md]: It'll automatically be turned into proper bullets in the rendered form. By leaving everything "1." you can insert a new point anywhere and all the numbering will automatically reflect the change :)

File: docs/features/generators.md (hunk 2 above)
Comment (agocke) [inline docs/features/generators.md]: Not really possible to know, without knowing the user's project file.

**Pull request merged**